x-pack/plugins/observability/server/services/slo/create_slo.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import uuid from 'uuid';
+
+import { SLO } from '../../types/models';
+import { ResourceInstaller } from './resource_installer';
+import { SLORepository } from './slo_repository';
+import { TransformInstaller } from './transform_installer';
+
+import { CreateSLOParams, CreateSLOResponse } from '../../types/schema';
+
+export class CreateSLO {
+  constructor(
+    private resourceInstaller: ResourceInstaller,
+    private repository: SLORepository,
+    private transformInstaller: TransformInstaller,
+    private spaceId: string
+  ) {}
+
+  public async execute(sloParams: CreateSLOParams): Promise<CreateSLOResponse> {
+    const slo = this.toSLO(sloParams);
+
+    await this.resourceInstaller.ensureCommonResourcesInstalled(this.spaceId);
```

Comment:
In case of an error thrown by the resourceInstaller, the application service will throw the error, and the [router will handle it](https://github.com/kdelemme/kibana/blob/main/x-pack/plugins/observability/server/routes/register_routes.ts#L74-L91).